src/services/notifier/api/notifierSvcProvider.ts

```diff
@@ -0,0 +1,154 @@
+import { ReturnObject, ServiceProvider } from '../../../httpClient'
+import { FetchError } from 'node-fetch'
+import { NotifierProviderError } from '../../../errors'
+import { ClientRequestArgs } from 'http'
+import { Status, Topic } from '../models/subscription.model'
+
+export type PlanStatusDTO =
+    | 'ACTIVE'
+
+export type PlanPriceDTO = {
+    price: string
+    currency: {
+        name: string
+        address: {
+            value: string
+            typeAsString: string
+        }
+    }
+}
+
+export type NotificationServiceTypeDTO = {
+  notificationServiceType: string
+  origin: string
+}
+
+export type NotifierChannel = {
+  name: string
+  source: string
+}
+
+export type SubscriptionPlanDTO = {
+    id: number
+    name: string
+    validity: number
+    planStatus: PlanStatusDTO
+    notificationPreferences: string[]
+    notificationQuantity: number
+    subscriptionPriceList: PlanPriceDTO[]
+}
+
+export type SubscriptionDTO = {
+    hash: string
+    currency: {
+      address: {
+        value: string
+      }
+    }
+    price: string
+    expirationDate: string
+    id: number
+    paid: boolean
+    status: Status
+    notificationBalance: number
+    subscriptionPlanId: number
+    previousSubscription?: SubscriptionDTO
+    topics: Array<Topic>
+    signature: string
+    userAddress: string
+    subscriptionPayments: any[]
+}
+
+export type NotifierResult<T> = {
+    message: | 'OK' // TODO: add the rest of the options
+    status: | 'OK' // TODO: add the rest of the options
+    content: T
+}
+
+export type ChannelMapper = (value: NotificationServiceTypeDTO, index: number, array: NotificationServiceTypeDTO[]) => NotifierChannel
+
+const ERROR_PFX = 'Notifier failed at endpoint'
+
+const RESOURCES = {
+  getSubscriptions: 'getSubscriptions',
+  getSubscriptionPlans: 'getSubscriptionPlans',
+  getAvailableChannels: 'getAvailableChannels'
```

Comment:
the endpoint is /info/availableNotificationPreferences

Reply:
ok